SAK-27900 - Allow numerical values for final grade override.

danielmerino opened the pull request.

ern commented on gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/CourseGradeOverridePanel.java:
```diff
@@ -100,10 +101,19 @@ public void onSubmit(final AjaxRequestTarget target, final Form<?> form) {
-						return;
+					
+					if (!schema.containsKey(newGrade))
+					{
```
Can we keep the formatting the same i.e. `if (something) {`

The brace styling with the brace on a separate line is not the preferred style!

Probably the most acceptable style is now google's checkstyle http://google.github.io/styleguide/javaguide.html

ern commented on gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/CourseGradeOverridePanel.java:
```diff
@@ -195,5 +205,36 @@ private String formatPoints(final CourseGrade courseGrade, final Gradebook grade
+		Double currentGradeValue = new Double(0.0);
+		try
+		{
+			Double dValue = Double.parseDouble(newGrade.replaceAll(",",".")); //The decimal separator is not relevant here.
```
For some reason when I see this comma/decimal replacing going on I cringe!

Shouldn't we just be using DecimalFormat? when dealing with input from other locales vs the string replacement?

danielmerino replied:
Sorry, @ern , I don't know how to use DecimalFormat to parse a value without knowing the locale and I'm not sure the locale is available here.

And in this case the comma replacement works, because users input is only used to get the closest lower final grade from the scale. It works even if user doesn't know his own locale and uses the wrong decimal separator. Though I admit that is not elegant neither readable.

But if you can share a sample of DecimalFormat ignoring commas, I will do the change.

danielmerino commented:
Hey, @ern I have changed the comma replacement for a pretty NumberFormat with user's locale. Also I have changed other things, please take a look if you can. Thanks!

ern commented:
Yeah this looks better! Though one day we should get this logic into a proper grading service!
